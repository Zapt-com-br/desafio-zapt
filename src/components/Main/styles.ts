import styled from 'styled-components'

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;

  background-color: #e9f5fe;
  color: #000000;
  padding: 3rem;
  text-align: center;
`

export const Logo = styled.img`
  width: 15rem;
  margin-bottom: 2rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
