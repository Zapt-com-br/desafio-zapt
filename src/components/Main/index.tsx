import * as S from './styles'

const Main = ({
  description = 'Junte seus amigos para comprar com desconto'
}) => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="Imagem do logo da Zapt" />
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default Main
