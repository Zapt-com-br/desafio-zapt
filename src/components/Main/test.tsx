import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the component correctly', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', {
        name: /junte seus amigos para comprar com desconto/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
