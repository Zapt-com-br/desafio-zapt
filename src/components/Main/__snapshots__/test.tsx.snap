// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  background-color: #e9f5fe;
  color: #000000;
  padding: 3rem;
  text-align: center;
}

.c1 {
  width: 15rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <img
    alt="Imagem do logo da Zapt"
    class="c1"
    src="/img/logo.svg"
  />
  <h2
    class="c2"
  >
    Junte seus amigos para comprar com desconto
  </h2>
</main>
`
